M2/Macaulay2/packages/NoetherianOperators.m2

```diff
@@ -0,0 +1,3649 @@
+-- -*- coding: utf-8 -*-
+newPackage(
+    "NoetherianOperators",
+    Version => "1.0",
+    Date => "Nov 25, 2020",
+    Authors => {
+        {Name => "Robert Krone", 
+        Email => "krone@math.gatech.edu"},
+        {Name => "Justin Chen", 
+        Email => "justin.chen@math.gatech.edu"},
+        {Name => "Marc Harkonen", 
+        Email => "harkonen@gatech.edu"},
+        {Name => "Yairon Cid-Ruiz",
+        Email => "Yairon.CidRuiz@UGent.be"},
+        {Name => "Anton Leykin",
+        Email => "anton.leykin@gmail.com"}
+    },
+    Headline => "numerically compute local dual spaces, Hilbert functions, and Noetherian operators",
+    PackageExports => {"Truncations", "Bertini", "NAGtypes"},
+    PackageImports => {"Dmodules", "PrimaryDecomposition"},
+    AuxiliaryFiles => false,
+    DebuggingMode => false,
+    Keywords => {"Numerical Algebraic Geometry", "Commutative Algebra"}
+)
+
+export {
+    "truncatedDual",
+    "zeroDimensionalDual",
+    "gCorners",
+    "localHilbertRegularity",
+    "eliminatingDual",
+    "innerProduct",
+    "reduceSpace",
+    "orthogonalInSubspace",
+    "Normalize",
+    "Rational",
+    "ProduceSB",
+    "numericalKernel",
+    "numericalImage",
+    "colReduce",
+
+    "DiffOp",
+    "ZeroDiffOp",
+    "InterpolatedDiffOp",
+    "diffOp",
+    "normalize",
+
+    "noetherianOperators",
+    "specializedNoetherianOperators",
+    "numericalNoetherianOperators",
+    "noethOpsFromComponents",
+    "coordinateChangeOps",
+    "rationalInterpolation",
+    "InterpolationTolerance",
+    "InterpolationDegreeLimit",
+    "NoetherianDegreeLimit",
+    "DependentSet",
+    "KernelStrategy",
+    "BM",
```

Comment:
Can you come up with a more descriptive name?

Reply:
Hi Dan, just to clarify: did you want a more descriptive name just for the option "BM", or the other ones as well? (Would something like "useBMStrategy" work?)

Reply:
My point was that the acronym "BM" is itself mysterious, so encapsulating it in something larger doesn't seem to be an improvement.

Reply:
@DanGrayson Would `IntegralStrategy` be an acceptable replacement for `BM` ?

Reply:
Yes, unless you expect the user to type `Strategy => IntegralStrategy`, in which case `Strategy => Integral` would look better.